src/Core/Checkout/Customer/Validation/Constraint/CustomerEmailUnique.php

```diff
@@ -28,27 +29,46 @@ class CustomerEmailUnique extends Constraint
     protected SalesChannelContext $salesChannelContext;
 
     /**
-     * @param array{salesChannelContext: SalesChannelContext} $options
+     * @param array{salesChannelContext?: SalesChannelContext}|null $options
+     *
+     * @deprecated tag:v6.8.0 - reason:new-optional-parameter - $options parameter will be removed, use $salesChannelContext instead
+     * @deprecated tag:v6.8.0 - reason:new-optional-parameter - $salesChannelContext will be required
      *
      * @internal
      */
-    public function __construct(array $options)
+    #[HasNamedArguments]
+    public function __construct(?array $options = null, ?SalesChannelContext $salesChannelContext = null)
     {
-        if (!($options['salesChannelContext'] ?? null) instanceof SalesChannelContext) {
-            throw CustomerException::missingOption('salesChannelContext', self::class);
+        if ($options !== null || $salesChannelContext === null) {
+            Feature::triggerDeprecationOrThrow(
+                'v6.8.0.0',
+                Feature::deprecatedMethodMessage(self::class, __METHOD__, 'v6.8.0.0', 'Use $salesChannelContext argument instead of providing it in $options array')
+            );
         }
 
-        if (!Feature::isActive('v6.8.0.0')) {
+        if ($options === null || Feature::isActive('v6.8.0.0')) {
+            parent::__construct();
+
+            if ($salesChannelContext === null) {
+                throw CustomerException::missingOption('salesChannelContext', self::class);
+            }
+
+            $this->salesChannelContext = $salesChannelContext;
+        } else {
+            if (!($options['salesChannelContext'] ?? null) instanceof SalesChannelContext) {
+                throw CustomerException::missingOption('salesChannelContext', self::class);
+            }
+
             if (!isset($options['context'])) {
                 $options['context'] = $options['salesChannelContext']->getContext();
             }
 
             if (!($options['context'] ?? null) instanceof Context) {
                 throw CustomerException::missingOption('context', self::class);
             }
-        }
 
-        parent::__construct($options);
+            parent::__construct($options);
+        }
     }
 
     /**
```

Comment:
why is this deprecated? feels unnecessary :thinking: you could also return `return $this->salesChannelContext->getContext();

Reply:
Ahh but the return type is changed so you need to deprecate it anyway. I find it unnecessary to have both `getContext` and `getSalesChannelContext`, which is just a shortcut to another